R/XcmsExperiment.R

```diff
@@ -1269,7 +1269,7 @@ setMethod(
             pkidx <- .i2index(peaks, rownames(.chromPeaks(object)), "peaks")
         else pkidx <- integer()
         res <- .mse_spectra_for_peaks(object, method, msLevel, expandRt,
-                                      expandMz, ppm, skipFilled, pkidx,
+                                      expandMz, ppm, skipFilled, unique(pkidx),
```

Comment:
Would anyone be surprised to get fewer (number of) peaks than requested ? Maybe a half-sentance in the manpage ?

Reply:
Actually, with this fix the number of peaks is identical to the requested number. The `unique()` here just ensures that we're not duplicating in two places - here we get the unique peaks and then later we duplicate and re-order them into what was requested from the user.